Week1.java

```diff
@@ -0,0 +1,54 @@
+package Homework;
+
+import java.util.HashMap;
+import java.util.Scanner;
+import java.lang.*;
+
+public class Week1 {
+	// validating the string for null and spaces. 
+	public static boolean ValidateString(String s) {
+		if(s==null) {
+			return false;
+		}
+		for(int i=0;i<s.length();i++) {
+			if(s.charAt(i)==' ') {
+				return false;
+			}
+		}
+		return true;
+	}
+	public static int LongestLengthSubstring(String s) {
+		int maxlength=0,i=0,count,j;
+		while(i<s.length()) {
+			count=0;
+			j=i;
+			HashMap<Character,Boolean> visited=new HashMap<>();
+			while(j<s.length() && !visited.containsKey(s.charAt(j))) {
+				visited.put(s.charAt(j), true);
+				count++;
+				j++;
+			}
+			// maxlength keeps the maximum length substring of non repeating characters till the current character i
+			if(count>maxlength) {
+				maxlength=count;
+			}
+			i++;

```

Comment:
looks like i is incremented one at a time and only at the end of the loop. That is usually an indicator that this while loop could be replaced with a for loop which is more clear when reading the code that we plan on going through every element of s